sdk/core/Azure.Core/src/TokenRequestContextFactory.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Contains the details of an authentication token request.
+    /// </summary>
+    public static class TokenRequestContextFactory
+    {
+        /// <summary>
+        /// Creates a <see cref="TokenRequestContext"/> from a dictionary.
+        /// </summary>
+        /// <param name="dictionary"></param>
+        /// <returns></returns>
+        /// <exception cref="System.ArgumentException"></exception>
+        public static TokenRequestContext FromDictionary(IReadOnlyDictionary<string, object> dictionary)
```

Comment:
Why does this take the dictionary and not nicely named and types parameters, e.g. scopes, claims, etc.

Reply:
I don't love this either but it seemed like the most generic way to allow for an arbitrary set of context to be passed to the provider. There doesn't seem to be any set of base properties that would make sense across various cloud auth schemes. I'd love some more ideas here.